### PR TITLE
feat(travis): add travis

### PR DESCRIPTION
**Pull request by itsachen**
## Context
Since we're in the process of publicizing the package we should move build automation from Circle , which is private to Lob, to Travis, which our other libraries currently use.

## What
- [x] Adds `.travis.yml`
  - Runs using Elixir `v1.4.5`, `v1.5.3` and `v1.6.1` and OTP `v20.0`
- [x] Adds build status badge to `README.md`
- [x] Enforces minimum coverage percentage
- [x] Upgrades `credo` because of bug encountered in Elixir `v1.6.1`
- [x] Ignores Dialyzer errors for meta-programmed resources

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -30,7 +31,7 @@ defmodule Lob.Mixfile do
   defp deps do
     [
       {:bypass, "~> 0.8", only: :test},
-      {:credo, "~> 0.8", only: [:dev, :test], runtime: false},
+      {:credo, "~> 0.9.0-rc1", only: [:dev, :test], runtime: false},
```
Comment (dmlittle) [inline mix.exs]: Any reason why we're using an RC dependency? I'd rather stick with 0.8 until 0.9 is released.

File: .dialyzer_ignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+lib/lob/intl_verification.ex:6: Function resource_url/1 will never be called
```
Comment (dmlittle) [inline .dialyzer_ignore]: I think we can do this for now. Given that `resource_url/1` is added through a macro, what do you think of adding an option to conditionally remove the `resource_url/1` definition (opt-in by default) such that we don't have these ignores to begin with.

```elixir
defmodule Lob.ResourceBase do

  defmacro __using__(opts) do
    options = Keyword.fetch!(opts, :options)

    quote do

      unless :ignore_resource_url in unquote(options), do
        @spec resource_url(String.t) :: String.t
        defp resource_url(resource_id), do: "#{base_url()}/#{resource_id}"
      end

    end

  end

end

defmodule Lob.IntlVerification do

  use Lob.ResourceBase, endpoint: "intl_verifications", methods: [], options: [:ignore_resource_url]

end

```

File: mix.exs (hunk 1 above)
Comment (itsachen) [reply mix.exs]: Ah I just chose the latest version recommended on the repo page. `0.8` has a bug with Elixir `1.6.x`, I can hunt around for another version.

Comment (itsachen) [reply mix.exs]: Sweet, I think `v0.8.10` fixed the bug.

File: .dialyzer_ignore (hunk 2 above)
Comment (itsachen) [reply .dialyzer_ignore]: Yeah that seems like a good compromise. I think its still a bit esoteric but its better than ignoring a legitimate warning.

Comment (itsachen) [reply .dialyzer_ignore]: Ticketed up as ENG-5695

**Pull request merged**